src/web/components/ClientSideTokenGeneration/CstgDomainsTable.tsx

```diff
@@ -0,0 +1,216 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { CheckedState } from '@radix-ui/react-checkbox';
+import { useState } from 'react';
+
+import { Dialog } from '../Core/Dialog';
+import { TriStateCheckbox, TriStateCheckboxState } from '../Core/TriStateCheckbox';
+
+import './CstgDomainsTable.scss';
+
+function NoApprovedDomains() {
+  return (
+    <div className='no-table-data-container'>
+      <img src='/group-icon.svg' alt='group-icon' />
+      <div className='no-table-data-text'>
+        <h2>No Approved Domains</h2>
+        <span>There are no approved domains.</span>
+      </div>
+    </div>
+  );
+}
+type CstgNewDomainRowProps = {
+  onAdd: (newDomain: string) => Promise<void>;
+};
+
+function CstgNewDomainRow({ onAdd }: CstgNewDomainRowProps) {
+  const [newDomain, setNewDomain] = useState<string>('');
+  return (
+    <tr>
+      <td />
+      <td className='domain'>
+        <input
+          className='input-container'
+          value={newDomain}
+          onChange={(e) => setNewDomain(e.target.value)}
+        />
+      </td>
+      <td className='action'>
+        <div className='action-cell'>
+          <button type='button' className='transparent-button' onClick={() => onAdd(newDomain)}>
+            Save
```

Comment:
Are you planning for hitting "Save" to be the only way to save a new row? I am thinking even just hitting enter or `onBlur` could possibly make the UX more smooth?

Reply:
In the next PR, I are going to introduce the validation and show a warning like "It does not look like, do you mean '....' " which the save button will be disabled and the user will need to click the correct value, I feel like it is better to let them click save button afterwards? Cause they can see the button is disabled and then turns into able after they correct the value. If using hitting enter, we maybe need to do more things to emphasis this action didn't go through?

Reply:
ahh good point, I forgot about validation. Sounds reasonable to leave as is for now then.